src/main/java/threads/ThreadsExecutor.java

```diff
@@ -0,0 +1,15 @@
+package threads;
+
+import server.CommunicatingServer;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class ThreadsExecutor implements MultiConnectionsExecutor {
+
+    @Override
+    public void execute(CommunicatingServer communicatingServer, int threadsNumber) {
+        ExecutorService threadPool = Executors.newFixedThreadPool(threadsNumber);
```

Comment:
This means that every time you want to exectue a *single* CommunicatingServer, you will create an entire new pool with up-to `threadNumber` of threads!

Reply:
True...